src/i18n/i18n-setup.ts

```diff
@@ -0,0 +1,46 @@
+import Vue from 'vue';
+import VueI18n from 'vue-i18n';
+import { Cookie } from '@nimiq/utils';
+
+Vue.use(VueI18n);
+
+const DEFAULT_LANGUAGE = 'en';
+const SUPPORTED_LANGUAGES = [DEFAULT_LANGUAGE];
+const LOADED_LANGUAGES: string[] = []; // our default language that is preloaded, if any
+
+export const i18n = new VueI18n({
+    locale: DEFAULT_LANGUAGE, // set locale (2 letters format: 'en')
+    fallbackLocale: DEFAULT_LANGUAGE, // fallback locale if no translation found
+    silentTranslationWarn: true, // disable the "no translation found" warning
+});
+
+function setI18nLanguage(lang: string) {
+    i18n.locale = lang;
+    return lang;
+}
+
+export function loadLanguageAsync(lang: string) {
+    if (!SUPPORTED_LANGUAGES.includes(lang)) lang = DEFAULT_LANGUAGE;
+
+    // If the language was already loaded
+    if (LOADED_LANGUAGES.includes(lang)) {
+        return Promise.resolve(setI18nLanguage(lang));
+    }
+
+    // If the language hasn't been loaded yet
+    return import(/* webpackChunkName: "lang-[request]" */ `@/i18n/${lang}.po`).then(
+        (messages) => {
+            i18n.setLocaleMessage(lang, messages.default);
+            LOADED_LANGUAGES.push(lang);
+            return setI18nLanguage(lang);
```

Comment:
Something like that :
```
router.beforeEach((to, from, next) =>
    loadLanguage(autodetectLanguage()).then(() => next()),
);
```

Reply:
Well, yes, something like this.
But `beforeEach` **does** delay fetching the view until the guard resolves.
You have to use [`beforeResolve`](https://router.vuejs.org/guide/advanced/navigation-guards.html#global-resolve-guards) instead.

Should also add a comment that this guard is to prevent switching to the new route before the language file finished loading.